packages/gatsby/src/redux/reducers/jobs.js

```diff
@@ -10,31 +10,27 @@ module.exports = (state = { active: [], done: [] }, action) => {
         throw new Error(`An ID must be provided when creating or setting job`)
       }
       const index = _.findIndex(state.active, j => j.id === action.payload.id)
-      const mergedJob = _.merge(state.active[index], {
-        ...action.payload,
-        createdAt: Date.now(),
-        plugin: action.plugin,
-      })
       if (index !== -1) {
+        const mergedJob = _.merge(state.active[index], {
+          ...action.payload,
+          createdAt: Date.now(),
+          plugin: action.plugin,
+        })
+
+        state.active[index] = mergedJob
         return {
```

Comment:
should we return just `state` if we are mutating it anyway?

Reply:
Oh good point, will do